core/templates/dev/head/pages/topics_and_skills_dashboard/TopicsListDirective.js

```diff
@@ -21,26 +21,45 @@ oppia.directive('topicsList', [
       restrict: 'E',
       scope: {
         getTopicSummaries: '&topicSummaries',
-        canDeleteTopic: '&userCanDeleteTopic'
+        canDeleteTopic: '&userCanDeleteTopic',
+        isInModal: '&inModal',
+        selectedTopicList: '='
       },
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/pages/topics_and_skills_dashboard/topics_list_directive.html'),
       controller: [
         '$scope', '$uibModal', '$rootScope', 'EditableTopicBackendApiService',
-        'AlertsService', 'EVENT_TOPIC_DELETED',
+        'AlertsService', 'EVENT_TOPICS_AND_SKILLS_DASHBOARD_REINITIALIZED',
         function(
             $scope, $uibModal, $rootScope, EditableTopicBackendApiService,
-            AlertsService, EVENT_TOPIC_DELETED) {
+            AlertsService, EVENT_TOPICS_AND_SKILLS_DASHBOARD_REINITIALIZED) {
           // As additional stories are not supported initially, it's not
           // being shown, for now.
           $scope.TOPIC_HEADINGS = [
             'name', 'subtopic_count', 'skill_count',
             'canonical_story_count', 'topic_status'
           ];
           $scope.getTopicEditorUrl = function(topicId) {
+            if ($scope.isInModal()) {
+              return;
+            }
             return '/topic_editor/' + topicId;
           };
 
+          $scope.selectTopic = function(topicId, topicVersion) {
+            if ($scope.selectedTopicList) {
+              if (
+                $scope.selectedTopicList.map(function(topic) {
+                  return topic.id;
+                }).indexOf(topicId) === -1) {
+                $scope.selectedTopicList.push({
```

Comment:
The "selected list" should contain minimal info. Can it just contain topic ID, so that you can call it selectedTopicIds or something like that?

Otherwise you should be defining a domain object instead of using dicts directly.

Reply:
For the backend api call that is done to add the skill to a topic, id and version of the topic are required. So, this is just an object with id and version of all topics.

Reply:
But that's still a "concept" that should be represented as a domain object. We don't want to pass arbitrary dicts around.

One thought -- maybe just store the topic IDs and, at the point of making the backend request, augment them with the versions as needed? I assume the versions are stored somewhere else (or we can make a dict mapping topic ID to version number). That would be better, I think.

Reply:
Done, the skills list has the editable topic summaries list. Now, taking version of topic from it and having only a topic id array.